File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        
        int n = nums.Length ;
        int [] result  = new int [n];
        int multi =1;
        int zerocount =0;
        for ( int i =0;i<n;i++){
            if ( nums[i]!=0){
                multi = multi* nums[i];
            }
            else {
                zerocount ++;
            }

        }
        for ( int  j =0;j<n;j++){
            if ( zerocount >1){
                result[j]=0;
            }
            else if ( zerocount==1){
                if (nums[j]==0){
                    result [j]= multi;
                }
                else {
                    result [j]= 0;
                }

            }
            else {
                result [j]= multi/ nums[j];
            }
        }
        return result ;

    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int zerocount =0;
        int totalproduct=1;
        int [] ans = new int [nums.length];
    
        for (int i =0;i<nums.length;i++){
            if (nums[i]==0){
                zerocount ++;
            }
            else {
                 totalproduct*=nums[i];

            }
           
        }
        for ( int i =0;i<nums.length ;i++){
            if (zerocount >1){
                ans[i]=0;
            }
            else if (zerocount==0){
                ans[i]=(totalproduct/nums[i]);
            }
            else {
                if (nums[i]==0){
                    ans[i]= totalproduct;
                }
                else{
                    ans[i]=0;
                }
            }
        }
        return ans ;
        
        
    }
}